### PR TITLE
Article : Insertion Sort

### PR DESCRIPTION
**Pull request by jainaman224**
#1048

### CODE REVIEW

**Comment by koustuvsinha** [comment]
Change the name of the file : 

```diff
- insertion-sort.md
+ Algorithms-Insertion-Sort.md
```

File: Algorithms-Insertion-Sort.md [hunk 1]
```diff
@@ -0,0 +1,113 @@
+# Insertion Sort
+
+Insertion sort is a [comparision based sorting](https://en.wikipedia.org/wiki/Comparison_sort). A sorting algorithm is comparison based, if it uses comparison operators (such as `less than` and `greated than`) to find the order between two numbers.
```
Comment (atjonathan) [inline Algorithms-Insertion-Sort.md]: ~~**comparision** based sorting~~ **comparison** based sorting

File: Algorithms-Insertion-Sort.md [hunk 2]
```diff
@@ -0,0 +1,113 @@
...
+
+In this sorting technique, we always maintain a sorted sublist in lower position of list and then we take one element from the rest of list and insert it at it's correct place. We does so till all elements are inserted into sublist. For example, while playing cards we sort cards in our hand. Starting from left and moving to right, we keep on inserting the card at it's right place till end.
```
Comment (atjonathan) [inline Algorithms-Insertion-Sort.md]: ~till all elements~~ **untill** all **of the** elements

File: Algorithms-Insertion-Sort.md [hunk 3]
```diff
@@ -0,0 +1,113 @@
...
+
+## Example
+
+![Insertion Sort](https://cloud.githubusercontent.com/assets/13117482/15633518/04ca4468-25cc-11e6-96af-feb395b456e0.png)
+
+In the above example, `grey shaded` sublist is always sorted. Please note that in the beginning, sublist contains ony one element, and *trivially* sorted. Then at each step we are inserting leftmost element of `white shaded` sublist at it's correct position.
```
Comment (atjonathan) [inline Algorithms-Insertion-Sort.md]: ~~`grey shaded`~~ **the** `grey shaded`

File: Algorithms-Insertion-Sort.md [hunk 4]
```diff
@@ -0,0 +1,113 @@
...
+```
+
+## Complexity
+
+```
+
+Space complexity: O(1)      // Auxillary/temporary space is used.
+
+Time complexity: O(n*n)     // Two nested for loops are used.
+
```
Comment (atjonathan) [inline Algorithms-Insertion-Sort.md]: Remove this line

File: Algorithms-Insertion-Sort.md [hunk 5]
```diff
@@ -0,0 +1,113 @@
...
+Loop for i=0 to N-1:
+* Pick element array[i] and insert it into sorted sublist array[0...i-1]
+
+```
+
+## Complexity
+
+```
+
```
Comment (atjonathan) [inline Algorithms-Insertion-Sort.md]: Remove this line

File: Algorithms-Insertion-Sort.md [hunk 6]
```diff
@@ -0,0 +1,113 @@
...
+Hence, we have sorted the complete list in this way.
+
+## Algorithm
+
+```
+
+Loop for i=0 to N-1:
+* Pick element array[i] and insert it into sorted sublist array[0...i-1]
+
```
Comment (atjonathan) [inline Algorithms-Insertion-Sort.md]: Remove this line

File: Algorithms-Insertion-Sort.md [hunk 7]
```diff
@@ -0,0 +1,113 @@
...
+![Insertion Sort](https://cloud.githubusercontent.com/assets/13117482/15633518/04ca4468-25cc-11e6-96af-feb395b456e0.png)
+
+In the above example, `grey shaded` sublist is always sorted. Please note that in the beginning, sublist contains ony one element, and *trivially* sorted. Then at each step we are inserting leftmost element of `white shaded` sublist at it's correct position.
+
+Hence, we have sorted the complete list in this way.
+
+## Algorithm
+
+```
+
```
Comment (atjonathan) [inline Algorithms-Insertion-Sort.md]: Remove this line

File: Algorithms-Insertion-Sort.md [hunk 8]
```diff
@@ -0,0 +1,113 @@
...
+Space complexity: O(1)      // Auxillary/temporary space is used.
+
+Time complexity: O(n*n)     // Two nested for loops are used.
+
+```
+
+## C++ Implementation
+
+```cpp
+
```
Comment (atjonathan) [inline Algorithms-Insertion-Sort.md]: Remove this line

File: Algorithms-Insertion-Sort.md [hunk 9]
```diff
@@ -0,0 +1,113 @@
...
+    // call the insertion sort
+    insertionSort(arr, 5);
+
+    // prints sorted array i.e. 5 6 11 12 13
+    for(int i=0; i<5; i++)
+        std::cout << arr[i] << " ";
+    return 0;
+}
+
```
Comment (atjonathan) [inline Algorithms-Insertion-Sort.md]: Remove this line

File: Algorithms-Insertion-Sort.md [hunk 10]
```diff
@@ -0,0 +1,113 @@
...
+        arr[j+1] = key # place element key at it's correct place
+
+# array to be sorted
+arr = [12, 11, 13, 5, 6]
+# call the insertion sort
+insertionSort(arr)
+# prints sorted array i.e. 5 6 11 12 13
+for i in range(len(arr)):
+    print(arr[i],end = ' ')
+
```
Comment (atjonathan) [inline Algorithms-Insertion-Sort.md]: Remove this line

File: Algorithms-Insertion-Sort.md [hunk 11]
```diff
@@ -0,0 +1,113 @@
...
+}
+
+```
+
+:rocket: [Run Code](https://repl.it/CWZq)
+
+## Python Implementation
+
+```python
+
```
Comment (atjonathan) [inline Algorithms-Insertion-Sort.md]: Remove this line

**Comment by koustuvsinha** [comment]
# LGTM :+1: 

Lets :package: :shipit:

**Comment by Rafase282** [comment]
LGTM :+1:

**Comment by alayek** [comment]
# LGTM :+1:  

Let's :shipit: :package:

**Pull request merged**